aggregation/gauge.go

```diff
@@ -20,20 +20,126 @@
 
 package aggregation
 
+import (
+	"math"
+
+	"github.com/m3db/m3metrics/policy"
+)
+
 var (
 	emptyGauge Gauge
 )
 
 // Gauge aggregates gauge values
 type Gauge struct {
-	value float64 // latest value received
+	last  float64
+	sum   float64
+	sumSq float64
+	count int64
+	max   float64
+	min   float64
+
+	customAggregationEnabled bool
 }
 
 // NewGauge creates a new gauge
-func NewGauge() Gauge { return emptyGauge }
+func NewGauge(useCustomeAggregation bool) Gauge {
+	if !useCustomeAggregation {
+		return emptyGauge
```

Comment:
I think this will return a copy of the empty gauge, which saves the allocation of making a new empty gauge and then copy it out.